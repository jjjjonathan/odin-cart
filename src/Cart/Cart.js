export default function Cart(props) {
  return (
    <div>
      <h3>Odincart contains:</h3>
      <div className="cart">
        {props.cartItems.map((item) => {
          return (
            <div className="cart-item" key={item.index}>
              <div className="cart-item-total-price">
                ${item.quantityInCart * item.price}
              </div>
              <div className="cart-item-title">{item.title}</div>
              <div className="cart-item-each-price">${item.price} each</div>
              <div className="cart-item-quantity">
                Qty: {item.quantityInCart}
                {/* <label htmlFor="quantity">Qty:</label>
                <select name="quantity" id="quantity">
                  <option value="1">1</option>
                  <option value="2">2</option>
                  <option value="3">3</option>
                  <option value="4">4</option>
                  <option value="5">5</option>
                  <option value="6">6</option>
                </select> */}
              </div>
            </div>
          );
        })}
        <div className="cart-total">
          Total: $
          {props.cartItems.reduce((accumulator, item) => {
            return accumulator + item.quantityInCart * item.price;
          }, 0)}
          <button type="button">Checkout</button>
        </div>
      </div>
    </div>
  );
}
