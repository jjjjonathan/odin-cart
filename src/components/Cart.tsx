import { ShopItem } from './Item';
import { RemoveFromCartHandler } from './CartStatus';

type CartProps = {
  items: ShopItem[];
  onQuantityChange: any; //fix
  onRemoveFromCart: RemoveFromCartHandler;
};

const Cart = ({ items, onQuantityChange, onRemoveFromCart }: CartProps) => {
  const cartItems = items
    .map((item, index) => ({ ...item, storeIndex: index }))
    .filter((item) => item.quantityInCart > 0);

  const cartTotal = () => {
    if (cartItems.length > 0) {
      return (
        <div className="cart-total">
          Total: $
          {cartItems.reduce((accumulator, item) => {
            return accumulator + item.quantityInCart * item.price;
          }, 0)}
          <button type="button">Checkout</button>
        </div>
      );
    } else {
      return null;
    }
  };

  const cartLead = () => {
    if (cartItems.length === 0) {
      return <p className="lead">nothing at the moment, apparently.</p>;
    } else {
      return null;
    }
  };

  return (
    <div>
      <h3>Odincart contains:</h3>
      {cartLead()}
      <div className="cart">
        {cartItems.map((item) => {
          return (
            <div className="cart-item" key={item.storeIndex}>
              <div className="cart-item-total-price">
                ${item.quantityInCart * item.price}
              </div>
              <div className="cart-item-title">{item.title}</div>
              <div className="cart-item-each-price">${item.price} each</div>
              <div className="cart-item-quantity">
                <label htmlFor="quantity">
                  Qty:
                  <select
                    name="quantity"
                    id="quantity"
                    value={item.quantityInCart}
                    onChange={onQuantityChange}
                    data-index={item.storeIndex}
                  >
                    <option value="1">1</option>
                    <option value="2">2</option>
                    <option value="3">3</option>
                    <option value="4">4</option>
                    <option value="5">5</option>
                    <option value="6">6</option>
                  </select>
                </label>
              </div>
              <button
                type="button"
                onClick={() => onRemoveFromCart(item.storeIndex)}
              >
                Remove from cart
              </button>
            </div>
          );
        })}
        {cartTotal()}
      </div>
    </div>
  );
};

export default Cart;
