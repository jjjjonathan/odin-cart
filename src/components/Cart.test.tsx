import renderer from 'react-test-renderer';
import Cart from './Cart';

const MockQuantityChange = jest.fn();
const MockRemoveFromCart = jest.fn();

const items = [
  {
    title: 'Odinfeathers',
    description: 'Faux feathers in pastel gradients',
    price: 595,
    quantityInCart: 3,
  },
  {
    title: 'The Nondescript Odinstone',
    description: 'If you expected description, See above',
    price: 9995,
    quantityInCart: 0,
  },
];

describe('Cart Component', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Cart
          items={items}
          onQuantityChange={MockQuantityChange}
          onRemoveFromCart={MockRemoveFromCart}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
