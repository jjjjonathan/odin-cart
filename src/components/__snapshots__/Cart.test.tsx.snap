// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component should match snapshot 1`] = `
<div>
  <h3>
    Odincart contains:
  </h3>
  <div
    className="cart"
  >
    <div
      className="cart-item"
    >
      <div
        className="cart-item-total-price"
      >
        $
        1785
      </div>
      <div
        className="cart-item-title"
      >
        Odinfeathers
      </div>
      <div
        className="cart-item-each-price"
      >
        $
        595
         each
      </div>
      <div
        className="cart-item-quantity"
      >
        <label
          htmlFor="quantity"
        >
          Qty:
          <select
            data-index={0}
            id="quantity"
            name="quantity"
            onChange={[MockFunction]}
            value={3}
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
          </select>
        </label>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Remove from cart
      </button>
    </div>
    <div
      className="cart-total"
    >
      Total: $
      1785
      <button
        type="button"
      >
        Checkout
      </button>
    </div>
  </div>
</div>
`;
